src/types/null.rs

```diff
@@ -0,0 +1,7 @@
+use crate::IonType;
+
+/// An in-memory representation of a typed `null`.
+// This type allows us to define custom behavior for `null` via trait implementations.
+// For an example, see the `WriteAsIonValue` trait.
+#[derive(Debug, PartialEq, Copy, Clone)]
+pub struct Null(pub IonType);
```

Comment:
🗺️ We have similar wrapper types in the `Element` API for `Blob(Bytes)`, `Clob(Bytes)`, `List(Sequence)` and `SExp(Sequence)`.